### DISCUSSION TITLE
Looking for aan approach to synchronise a heavy state in chunks by node's request

### DISCUSSION THREAD

**Discussion by sammy007** [libp2p/rust-libp2p#5640] [Q&A]
I am looking for a pretty similar approach widely used in blockchains - chain sync.
Once upon a time node realizes it is not in sync with other nodes. So it must request a sync from a network. It's fine if one random node will fulfill request by launching a long running process and pushing its state in chunks (there are blocks, so block after block or N blocks a time) to a peer which requested the sync.

What first comes to my mind: request-response to a random peer, request sync, get response with N blocks. But it's too dead simple solution it seems.

What would community suggest?

**Comment by drHuangMHT** [comment]
For reaching out to other nodes you can use `libp2p-kad`. As for synchronization you can take a look at `libp2p-gossipsub`, which includes mechanisms for catching up with historical messages in the network. You can of course design a protocol yourself.
